tests/test_vb.py

```diff
@@ -48,58 +31,67 @@ def __init__(self, forward_model, data):
         self._forward_model = forward_model
         self._data = data
 
-        if not hasattr(forward_model, "jacobian"):
-            delattr(self, "jacobian")
-
     def __call__(self, parameters):
         model = self._forward_model(parameters)
-        repetitions = self._data.shape[0] // model.shape[0]
+        errors = []
+        for data in self._data:
+            errors.append(model - data)
+
+        return {"noise0": np.concatenate(errors)}
 
-        return self._data - stack_data(model, repetitions)
 
-class ModelErrorWithJacobian(ModelError): 
+class ModelErrorWithJacobian(ModelError):
     def jacobian(self, parameters):
-        jac = self._forward_model.jacobian(parameters)
-        repetitions = self._data.shape[0] // jac.shape[0]
-        return stack_data(jac, repetitions)
+        jac = -self._forward_model.jacobian(parameters)
+        full_jac = np.tile(jac, (len(self._data), 1))
+        return {"noise0": full_jac}
 
 
 class Test_VB(unittest.TestCase):
-
     def run_vb(self, n_data, given_jac=False, plot=False):
         np.random.seed(6174)
 
         fw = ForwardModel()
-        param_true = (7., 10.)
-        noise_std = 0.1
+        param_true = [7.0, 10.0]
+        noise_sd = 0.1
 
-        data = stack_data(fw(param_true), n_data, noise_std)
+        data = []
+        perfect_data = fw(param_true)
+        for _ in range(n_data):
+            data.append(perfect_data + np.random.normal(0, noise_sd, len(perfect_data)))
 
         if given_jac:
             me = ModelErrorWithJacobian(fw, data)
         else:
             me = ModelError(fw, data)
 
         param_prior = MVN([6, 11], [[1 / 3 ** 2, 0], [0, 1 / 3 ** 2]])
-        noise_prior = Gamma(s=0.1, c=1000)
+        noise_prior = {"noise0": Gamma.FromSD(3*noise_sd)}
```

Comment:
You mean "FromSD"?

Reply:
yes, I have the impression that this hides what is in there from the user. Instead, there could be some functions that e.g. return the parameters, but it should always be clear that we are providing shape and scale instead of std.

Reply:
I answered in #16 to not clutter this PR.